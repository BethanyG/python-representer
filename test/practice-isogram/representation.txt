Module([FunctionDef('placeholder_0', arguments([], [arg('placeholder_1', lineno=1, col_offset=15, end_lineno=1, end_col_offset=21)], kwonlyargs=[], kw_defaults=[], defaults=[]), [Assign([Name('placeholder_2', Store(), lineno=2, col_offset=4, end_lineno=2, end_col_offset=20)], ListComp(Call(Attribute(Name('placeholder_3', Load(), lineno=2, col_offset=24, end_lineno=2, end_col_offset=28), 'lower', Load(), lineno=2, col_offset=24, end_lineno=2, end_col_offset=34), [], [], lineno=2, col_offset=24, end_lineno=2, end_col_offset=36), [comprehension(Name('placeholder_3', Store(), lineno=2, col_offset=41, end_lineno=2, end_col_offset=45), Name('placeholder_1', Load(), lineno=2, col_offset=49, end_lineno=2, end_col_offset=55), [Call(Attribute(Name('placeholder_3', Load(), lineno=2, col_offset=59, end_lineno=2, end_col_offset=63), 'isalpha', Load(), lineno=2, col_offset=59, end_lineno=2, end_col_offset=71), [], [], lineno=2, col_offset=59, end_lineno=2, end_col_offset=73)], 0)], lineno=2, col_offset=23, end_lineno=2, end_col_offset=74), lineno=2, col_offset=4, end_lineno=2, end_col_offset=74), Return(Compare(Call(Name('len', Load(), lineno=3, col_offset=11, end_lineno=3, end_col_offset=14), [Call(Name('set', Load(), lineno=3, col_offset=15, end_lineno=3, end_col_offset=18), [Name('placeholder_2', Load(), lineno=3, col_offset=19, end_lineno=3, end_col_offset=35)], [], lineno=3, col_offset=15, end_lineno=3, end_col_offset=36)], [], lineno=3, col_offset=11, end_lineno=3, end_col_offset=37), [Eq()], [Call(Name('len', Load(), lineno=3, col_offset=41, end_lineno=3, end_col_offset=44), [Name('placeholder_2', Load(), lineno=3, col_offset=45, end_lineno=3, end_col_offset=61)], [], lineno=3, col_offset=41, end_lineno=3, end_col_offset=62)], lineno=3, col_offset=11, end_lineno=3, end_col_offset=62), lineno=3, col_offset=4, end_lineno=3, end_col_offset=62)], [], lineno=1, col_offset=0, end_lineno=3, end_col_offset=62)], [])