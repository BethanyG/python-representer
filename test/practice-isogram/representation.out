## BEGIN NORMALIZED CODE ##
def placeholder_0(placeholder_1):
    placeholder_2 = [
        placeholder_3.lower()
        for placeholder_3 in placeholder_1
        if placeholder_3.isalpha()
    ]
    return len(set(placeholder_2)) == len(placeholder_2)

## END NORMALIZED CODE ##

## BEGIN TREE BEFORE ##
Module(
  [
    FunctionDef(
      'is_isogram',
      arguments(
        [],
        [
          arg(
            'string',
            lineno=1,
            col_offset=15,
            end_lineno=1,
            end_col_offset=21)],
        kwonlyargs=[],
        kw_defaults=[],
        defaults=[]),
      [
        Assign(
          [
            Name(
              'characters_lower',
              Store(),
              lineno=2,
              col_offset=4,
              end_lineno=2,
              end_col_offset=20)],
          ListComp(
            Call(
              Attribute(
                Name(
                  'char',
                  Load(),
                  lineno=2,
                  col_offset=24,
                  end_lineno=2,
                  end_col_offset=28),
                'lower',
                Load(),
                lineno=2,
                col_offset=24,
                end_lineno=2,
                end_col_offset=34),
              [],
              [],
              lineno=2,
              col_offset=24,
              end_lineno=2,
              end_col_offset=36),
            [
              comprehension(
                Name(
                  'char',
                  Store(),
                  lineno=2,
                  col_offset=41,
                  end_lineno=2,
                  end_col_offset=45),
                Name(
                  'string',
                  Load(),
                  lineno=2,
                  col_offset=49,
                  end_lineno=2,
                  end_col_offset=55),
                [
                  Call(
                    Attribute(
                      Name(
                        'char',
                        Load(),
                        lineno=2,
                        col_offset=59,
                        end_lineno=2,
                        end_col_offset=63),
                      'isalpha',
                      Load(),
                      lineno=2,
                      col_offset=59,
                      end_lineno=2,
                      end_col_offset=71),
                    [],
                    [],
                    lineno=2,
                    col_offset=59,
                    end_lineno=2,
                    end_col_offset=73)],
                0)],
            lineno=2,
            col_offset=23,
            end_lineno=2,
            end_col_offset=74),
          lineno=2,
          col_offset=4,
          end_lineno=2,
          end_col_offset=74),
        Return(
          Compare(
            Call(
              Name(
                'len',
                Load(),
                lineno=3,
                col_offset=11,
                end_lineno=3,
                end_col_offset=14),
              [
                Call(
                  Name(
                    'set',
                    Load(),
                    lineno=3,
                    col_offset=15,
                    end_lineno=3,
                    end_col_offset=18),
                  [
                    Name(
                      'characters_lower',
                      Load(),
                      lineno=3,
                      col_offset=19,
                      end_lineno=3,
                      end_col_offset=35)],
                  [],
                  lineno=3,
                  col_offset=15,
                  end_lineno=3,
                  end_col_offset=36)],
              [],
              lineno=3,
              col_offset=11,
              end_lineno=3,
              end_col_offset=37),
            [
              Eq()],
            [
              Call(
                Name(
                  'len',
                  Load(),
                  lineno=3,
                  col_offset=41,
                  end_lineno=3,
                  end_col_offset=44),
                [
                  Name(
                    'characters_lower',
                    Load(),
                    lineno=3,
                    col_offset=45,
                    end_lineno=3,
                    end_col_offset=61)],
                [],
                lineno=3,
                col_offset=41,
                end_lineno=3,
                end_col_offset=62)],
            lineno=3,
            col_offset=11,
            end_lineno=3,
            end_col_offset=62),
          lineno=3,
          col_offset=4,
          end_lineno=3,
          end_col_offset=62)],
      [],
      lineno=1,
      col_offset=0,
      end_lineno=3,
      end_col_offset=62)],
  [])
## END TREE BEFORE ##

## BEGIN NORMALIZED TREE ##
Module(
  [
    FunctionDef(
      'placeholder_0',
      arguments(
        [],
        [
          arg(
            'placeholder_1',
            lineno=1,
            col_offset=15,
            end_lineno=1,
            end_col_offset=21)],
        kwonlyargs=[],
        kw_defaults=[],
        defaults=[]),
      [
        Assign(
          [
            Name(
              'placeholder_2',
              Store(),
              lineno=2,
              col_offset=4,
              end_lineno=2,
              end_col_offset=20)],
          ListComp(
            Call(
              Attribute(
                Name(
                  'placeholder_3',
                  Load(),
                  lineno=2,
                  col_offset=24,
                  end_lineno=2,
                  end_col_offset=28),
                'lower',
                Load(),
                lineno=2,
                col_offset=24,
                end_lineno=2,
                end_col_offset=34),
              [],
              [],
              lineno=2,
              col_offset=24,
              end_lineno=2,
              end_col_offset=36),
            [
              comprehension(
                Name(
                  'placeholder_3',
                  Store(),
                  lineno=2,
                  col_offset=41,
                  end_lineno=2,
                  end_col_offset=45),
                Name(
                  'placeholder_1',
                  Load(),
                  lineno=2,
                  col_offset=49,
                  end_lineno=2,
                  end_col_offset=55),
                [
                  Call(
                    Attribute(
                      Name(
                        'placeholder_3',
                        Load(),
                        lineno=2,
                        col_offset=59,
                        end_lineno=2,
                        end_col_offset=63),
                      'isalpha',
                      Load(),
                      lineno=2,
                      col_offset=59,
                      end_lineno=2,
                      end_col_offset=71),
                    [],
                    [],
                    lineno=2,
                    col_offset=59,
                    end_lineno=2,
                    end_col_offset=73)],
                0)],
            lineno=2,
            col_offset=23,
            end_lineno=2,
            end_col_offset=74),
          lineno=2,
          col_offset=4,
          end_lineno=2,
          end_col_offset=74),
        Return(
          Compare(
            Call(
              Name(
                'len',
                Load(),
                lineno=3,
                col_offset=11,
                end_lineno=3,
                end_col_offset=14),
              [
                Call(
                  Name(
                    'set',
                    Load(),
                    lineno=3,
                    col_offset=15,
                    end_lineno=3,
                    end_col_offset=18),
                  [
                    Name(
                      'placeholder_2',
                      Load(),
                      lineno=3,
                      col_offset=19,
                      end_lineno=3,
                      end_col_offset=35)],
                  [],
                  lineno=3,
                  col_offset=15,
                  end_lineno=3,
                  end_col_offset=36)],
              [],
              lineno=3,
              col_offset=11,
              end_lineno=3,
              end_col_offset=37),
            [
              Eq()],
            [
              Call(
                Name(
                  'len',
                  Load(),
                  lineno=3,
                  col_offset=41,
                  end_lineno=3,
                  end_col_offset=44),
                [
                  Name(
                    'placeholder_2',
                    Load(),
                    lineno=3,
                    col_offset=45,
                    end_lineno=3,
                    end_col_offset=61)],
                [],
                lineno=3,
                col_offset=41,
                end_lineno=3,
                end_col_offset=62)],
            lineno=3,
            col_offset=11,
            end_lineno=3,
            end_col_offset=62),
          lineno=3,
          col_offset=4,
          end_lineno=3,
          end_col_offset=62)],
      [],
      lineno=1,
      col_offset=0,
      end_lineno=3,
      end_col_offset=62)],
  [])
## END NORMALIZED TREE ##